File: RukaLjubavi.Api/Models/Donacija.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RukaLjubavi.Api.Models
{ 
    public enum StatusDonacije
    {
        Aktivna,
        Na_cekanju,
        Prihvacena,
        Odbijena,
        U_toku,
        Zavrsena
    }
    public class Donacija : IEntity
    {
        [Key]
        public int Id { get; set; }
        public string Opis { get; set; }
        public int? Kolicina { get; set; }
        public StatusDonacije StatusDonacije { get; set; }
        public DateTime? DatumVrijeme { get; set; }

        [ForeignKey("BenefiktorId,KategorijaId")]
        public virtual BenefiktorKategorija BenefiktorKategorije { get; set; }
        public int? BenefiktorId { get; set; }
        public int? KategorijaId { get; set; }
        public string NazivKategorije { get; set; }

        [ForeignKey(nameof(DonatorId))]
        public virtual Donator Donator { get; set; }
        public int? DonatorId { get; set; }
    }
}

File: RukaLjubavi.Android/app/src/main/java/com/fit/ba/rukaljubavi/Models/Donacija.kt
package com.fit.ba.rukaljubavi.Models

import java.io.Serializable

class Donacija(
    var id: Int,
    var opis: String,
    var kolicina: Int,
    var datumVrijeme: String,
    var donatorId: Int,
    var donatorIme: String,
    var donatorPrezime: String,
    var benefiktorId: Int,
    var benefiktorNazivKompanije: String,
    var benefiktorLokacija: String,
    var nazivKategorije: String,
    var status: String
) : Serializable {
}